Refactor how we handle "mixed" commits

jazzsequence opened the pull request:
Removes a bunch of failing code handling "mixed" commits that wasn't actually running except when it was failing and breaking things anyway.

"Mixed" commits fail the checks run on PRs, appropriately, and this code was never actually adding those commits to the array of commits to deploy. It was built to solve a problem where mixed commits would fail the deploy process, which we wanted to avoid. If the goal was "skipping" the commit, the way we can do that is just not adding it to the array of commits to deploy.

jazzsequence committed:
Various minor improvements
| This update includes minor improvements to the Sage install script, the Composer environment and includes backported updates to Bedrock itself.

pwtyler commented on devops/scripts/deploy-public-upstream.sh:
```diff
@@ -33,19 +33,13 @@ for commit in $newcommits; do
-        fi
-      done
-    done
+    continue
```
```suggestion
```
with nothing below this if block, no continue necessary.

jazzsequence committed:
don't need the continue since we're at the end of the for

Co-authored-by: Phil Tyler <philip@tylerdigital.com>

jazzsequence committed:
Various minor improvements
| This update includes minor improvements to the Sage install script, the Composer environment and includes backported updates to Bedrock itself.

jazzsequence committed:
Various minor improvements

| This update includes minor improvements to the Sage install script, the Composer environment and includes backported updates to
Bedrock itself.
